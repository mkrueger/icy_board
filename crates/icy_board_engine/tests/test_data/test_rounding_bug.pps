REAL A
INTEGER B
B = 5
A = B
PRINTLN STRING(A/2)

B = 5 
IF (TOREAL(B) / 2 = B / 2) THEN
    PRINTLN "Even 1"
ENDIF

B = 6
IF (TOREAL(B) / 2 = B / 2) THEN
    PRINTLN "Even 2"
ENDIF
PRINTLN S2I("123", 10), " ", S2I("12FOO", 10), " ", S2I("_I", 10), " ", S2I("0FOOBAR", 10), " ", S2I("0.0", 10)

PRINTLN !"0", !"43423", !"0000", !"0.0", !"0FOOBAR", !"1.1", !"_I", !32768, !32767
