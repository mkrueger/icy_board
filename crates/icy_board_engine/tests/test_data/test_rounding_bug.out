2.5
Even 2
123 12 0 0 0
101110100
